File: services/core/entity/habit_input.go
package entity

import "tenkhours/pkg/types"

type HabitInput struct {
	ID             *string        `json:"id,omitempty"`
	CategoryID     *string        `json:"categoryID"`
	CompletionType CompletionType `json:"completionType"`
	Name           string         `json:"name"`
	Value          float64        `json:"value"`
	Unit           *string        `json:"unit"`
	RRule          string         `json:"rrule"`
	ResetDuration  HabitReset     `json:"resetDuration"`
}

type HabitPipeline struct {
	Filter  *HabitFilter
	OrderBy *HabitOrderBy
	*types.Pagination
}

type HabitFilter struct {
	CharacterID  *string  `json:"character_id"`
	CharacterIDs []string `json:"character_ids"`
}

type HabitOrderBy struct{}

File: services/penk/src/utils/grpc/proto/core/HabitInput.ts
// Original file: ../../proto/core/habit_message.proto

import type { CompletionType as _core_CompletionType, CompletionType__Output as _core_CompletionType__Output } from '../core/CompletionType';
import type { HabitReset as _core_HabitReset, HabitReset__Output as _core_HabitReset__Output } from '../core/HabitReset';

export interface HabitInput {
  'id'?: (string);
  'categoryId'?: (string);
  'completionType'?: (_core_CompletionType);
  'name'?: (string);
  'value'?: (number | string);
  'unit'?: (string);
  'rrule'?: (string);
  'resetDuration'?: (_core_HabitReset);
  '_id'?: "id";
  '_categoryId'?: "categoryId";
  '_unit'?: "unit";
}

export interface HabitInput__Output {
  'id'?: (string);
  'categoryId'?: (string);
  'completionType': (_core_CompletionType__Output);
  'name': (string);
  'value': (number);
  'unit'?: (string);
  'rrule': (string);
  'resetDuration': (_core_HabitReset__Output);
  '_id': "id";
  '_categoryId': "categoryId";
  '_unit': "unit";
}
